contracts/staking/PoSValidatorManager.sol

```diff
@@ -92,34 +90,36 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         uint32 messageIndex
     ) external {
         if (includeUptimeProof) {
-            PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
-            (WarpMessage memory warpMessage, bool valid) =
-                WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
-            require(valid, "PoSValidatorManager: invalid warp message");
-
-            require(
-                warpMessage.sourceChainID == WARP_MESSENGER.getBlockchainID(),
-                "PoSValidatorManager: invalid source chain ID"
-            );
-            require(
-                warpMessage.originSenderAddress == address(0),
-                "PoSValidatorManager: invalid origin sender address"
-            );
-
-            (bytes32 uptimeValidationID, uint64 uptime) =
-                ValidatorMessages.unpackValidationUptimeMessage(warpMessage.payload);
-            require(
-                validationID == uptimeValidationID,
-                "PoSValidatorManager: invalid uptime validation ID"
-            );
-
-            $._validatorUptimes[validationID] = uptime;
-            emit ValidationUptimeUpdated(validationID, uptime);
+            _getUptime(validationID, messageIndex);
         }
+        // TODO: Calculate the reward for the validator, but do not unlock it
 
         _initializeEndValidation(validationID);
     }
 
+    function _getUptime(bytes32 validationID, uint32 messageIndex) internal view returns (uint64) {
+        (WarpMessage memory warpMessage, bool valid) =
+            WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
+        require(valid, "PoSValidatorManager: invalid warp message");
+
+        require(
+            warpMessage.sourceChainID == WARP_MESSENGER.getBlockchainID(),
+            "PoSValidatorManager: invalid source chain ID"
+        );
+        require(
+            warpMessage.originSenderAddress == address(0),
+            "PoSValidatorManager: invalid origin sender address"
+        );
+
+        (bytes32 uptimeValidationID, uint64 uptime) =
```

Comment:
Just for my own understanding, does uptime return the absolute amount of time that the validator was up, or a percent of time that it was up during the staking period? I'm just thinking of whether other information is necessary to calculate rewards.

Reply:
It returns the absolute number of seconds the validator has been up